### ISSUE TITLE
Why modify value object of a entry only if equals() returns false? 

### ISSUE THREAD

**Issue by praveen12bnitt** [jhalterman/expiringmap#12]
This is not a bug, but a question.

Is there a reason why you are storing the value of a entry only if equals() method returns false

https://github.com/jhalterman/expiringmap/blob/master/src/main/java/net/jodah/expiringmap/ExpiringMap.java#L1106

This created some problem for me when i attempted to integrate it with https://github.com/spring-projects/spring-session

I am using your library, so that the session are automatically removed from a map, after the session timeout happens.
```java
Map<String,ExpiringSession> sessions = ExpiringMap.builder()
				  .expirationPolicy(ExpirationPolicy.ACCESSED)
				  .expiration(maxInactiveIntervalInSeconds, TimeUnit.SECONDS)
				  .expirationListener(new ExpirationListener<String, ExpiringSession>() {
					@Override
					public void expired(String key, ExpiringSession value) {
						logger.error("Session {} expired. So removing it from memory", key);
					}
				})
				  .build(); 
```

In short, spring adds MapSession[1] data to the map, after every request is completed. If you see the implementation of MapSession, the equals method is based on the session id. 

Now when i use your map with spring session, since the equal method says the value object has not changed, you dont modify anything and as a result of it, the next request for getSession() does not return a most up-to-date session object.

If i replace your map with JDK's ConcurrentHashMap, thing work fine since the JDK implementation does not have a equals check. 

Based on your response, i will submit a pull request to enable/disable this behavior using the builder. 

[1] https://github.com/spring-projects/spring-session/blob/master/spring-session/src/main/java/org/springframework/session/MapSession.java

**Comment by jhalterman** [comment]
I just updated some code here to improve this a bit, but the intent of this was when it comes to rescheduling we should ignore `put` calls that do not change the value for a key. That is *unless* the ExpirationPolicy is `ACCESSED` in which case all `put` calls should reschedule expiration.

I believe the commit I just made should fix your issue (if I understand it correctly). Let me know if it does not:

https://github.com/jhalterman/expiringmap/commit/57abc13783dcb2ba636a9024014d7081adc8e0b2#diff-dce92a6a312d0d78dd21fab98a173b4cR1120

**Comment by praveen12bnitt** [comment]
Your understanding of the problem is accurate. The changes that you pushed did fix the problem that I had. Do you plan to make a release with this change anytime soon? This will help me plan my build.

**Comment by praveen12bnitt** [comment]
Thanks Jonathan Halterman :)
